### PR TITLE
Builder and pathing adjustments

### LINKED ISSUE #6654: Builder does not start building
- [ x] I am running the latest **alpha** version of MineColonies for my Minecraft version.
- [ x] I made sure that **this issue is not a duplicate of any existing issue.**

- Minecraft Version: 1.16.5
- MineColonies Version: 0.13.669-alpha

## Steps to reproduce the issue / details
One of my builder does not start building after recieving an order. The chat never says: "(builder) has started building (building name)" I have tried to fire and rehire him. Fired him and hired another worker. The issue seems to affect whatever builder that uses that specific builders hut and whatever building he is trying to build. My other builder works just fine. 

The log says:
[19:29:59] [Server thread/INFO] [minecolonies/]: WorkOrderBuild at location BlockPos{x=376, y=74, z=879} is using schematics/caledonia/plantation1 instead of schematics/caledonia/plantation1
19:29:59.940
[19:29:59] [Render thread/INFO] [minecraft/NewChatGui]: [CHAT] Build request created!

So the request is created, but the building doesn't start and the builder just wanders around without requesting any item or doing anything useful.

Here is more of the log: https://paste.ubuntu.com/p/Jznq5y5N7f/

### PR DESCRIPTION
**Pull request by someaddons**
Closes #6654

# Changes proposed in this pull request:
Fix builder pathing issues

Restructured where our pathing queries are held, instead of having query(job), temporary resultholder(Pathresult), calculation future and the result(path) beeing split data and making it hard to line up which matches which they're all now bundled together. The pathresults holds both the Job, the calculating future aswell as its result data, the actual Path. This change also allows us to easily use pathfinding outside of a navigator/ion purpose, e.g. for finding a nice raider spawnpoint.

Review please

### CODE REVIEW

**Comment by Raycoms** [comment]
:shipit: looks good

File: src/api/java/com/minecolonies/api/entity/pathfinding/AbstractAdvancedPathNavigate.java [hunk 1]
```diff
@@ -107,7 +103,7 @@ public BlockPos getDestination()
      * @param speed the speed to run at.
      * @return the result of the pathing.
      */
-    public abstract RandomPathResult moveToRandomPos(final double range, final double speed);
+    public abstract PathResult moveToRandomPos(final double range, final double speed);
```
Comment (marchermans) [inline src/api/java/com/minecolonies/api/entity/pathfinding/AbstractAdvancedPathNavigate.java:106]: Please backport this into PVI

**Review by marchermans — APPROVED** [review]
Generally good additions. 
Could you backport this into PVI as well?

File: src/api/java/com/minecolonies/api/entity/pathfinding/AbstractAdvancedPathNavigate.java (hunk 1 above)
Comment (someaddons) [reply src/api/java/com/minecolonies/api/entity/pathfinding/AbstractAdvancedPathNavigate.java:106]: ye intended to do that aswell

**Pull request merged**